base/toml_parser.jl

```diff
@@ -658,9 +658,12 @@ end
 
 function push!!(v::Vector, el)
     T = eltype(v)
-    if el isa T || typeof(el) === T
+    t = typeof(el)
+    if el isa T || t === T
         push!(v, el::T)
         return v
+    elseif T === Union{}
```

Comment:
Will help inference not enter into a bad frame of `copy!(new, v::Vector{Union{}})`.